server/src/main/java/org/elasticsearch/action/search/MultiSearchRequest.java

```diff
@@ -276,10 +276,9 @@ public static void readMultiLineFormat(BytesReference data,
     }
 
     private static int findNextMarker(byte marker, int from, BytesReference data, int length) {
-        for (int i = from; i < length; i++) {
-            if (data.get(i) == marker) {
-                return i;
-            }
+        final int res = data.findNextMarker(marker, from, length);
+        if (res > -1) {
```

Comment:
this should be `!= -1` and assert that it's positive instead?

Reply:
Right, adjusted accordingly :)